prime-router/src/main/kotlin/history/SubmissionHistory.kt

```diff
@@ -303,17 +313,71 @@ class DetailedSubmissionHistory(
     }
 
     /**
-     * Enrich a parent detailed history with details from the process or translate action.
+     * Enrich a parent detailed history with details from the route and translate actions.
+     * Add destinations, errors, and warnings, to the history details.
+     * Note: Route/Translate is exclusive to the Universal pipeline
+     * See enrichWithProcessAction for the COVID pipeline counterpart
+     *
+     * @param descendants[] route and translate actions that will be used to enrich
+     */
+    private fun enrichWithRoutingAndTranslationActions(descendants: List<DetailedSubmissionHistory>) {
+        require(topic == "full-elr") {
+            "Must be route or translate actions for a full-elr submission"
+        }
+
+        val filterLogs = mutableListOf<DetailedActionLog>()
+
+        descendants.filter { it.actionName == TaskAction.route }.forEach { descendant ->
```

Comment:
Add some more comments, so we can understand this

Reply:
done